locator/token_metadata.hh

```diff
@@ -72,6 +72,50 @@ struct host_id_or_endpoint {
 class token_metadata_impl;
 struct topology_change_info;
 
+struct version_tracker {
+public:
+    friend class shared_token_metadata;
+
+    using link_base = boost::intrusive::list_member_hook<boost::intrusive::link_mode<boost::intrusive::auto_unlink>>;
+    struct link_type : link_base {
+        link_type() noexcept = default;
+        link_type(link_type&& o) noexcept { swap_nodes(o); }
+        link_type& operator=(link_type&& o) noexcept {
+            if (this != &o) {
+                this->~link_type();
+                new(this) link_type(std::move(o));
+            }
+            return *this;
+        }
+    };
+private:
+    utils::phased_barrier::operation _op;
+    service::topology::version_t _version;
+    link_type _link;
+
+    // When version is marked as stalling, this is engaged with the barrier start time
+    // so that we can calculate accurate stall time.
+    std::optional<std::chrono::steady_clock::time_point> _stalled;
+public:
+    version_tracker() = default;
```

Comment:
Seems unneeded.

Reply:
The code doesn't compile without it, there are some places where tracker is default-initialized.